### PR TITLE
Refactor page size calculation to use Short.toUnsignedInt for clarity.

### PR DESCRIPTION
**Pull request by rohitpaulk**

### CODE REVIEW

**Comment by rohitpaulk** [comment]
@andy1li can you review this, let me know if (a) it is "correct" (b) whether we can do the same in the Kotlin starter too?

File: compiled_starters/java/src/main/java/Main.java [hunk 1]
```diff
@@ -20,8 +20,8 @@ public static void main(String[] args){
           byte[] header = Files.readAllBytes(Path.of(databaseFilePath));
 
           // The page size is stored at the 16th byte offset, using 2 bytes in big-endian order.
-          // '& 0xFFFF' is used to convert the signed short to an unsigned int.
-          int pageSize = ByteBuffer.wrap(header).order(ByteOrder.BIG_ENDIAN).position(16).getShort() & 0xFFFF;
+          short pageSizeSigned = ByteBuffer.wrap(header).order(ByteOrder.BIG_ENDIAN).position(16).getShort();
```
Comment (andy1li) [inline compiled_starters/java/src/main/java/Main.java:23]: Do we need `.order(ByteOrder.BIG_ENDIAN)`? 
[Doc](https://docs.oracle.com/javase/8/docs/api/java/nio/ByteBuffer.html) says that ByteBuffer is BIG_ENDIAN by default:
> The initial order of a byte buffer is always BIG_ENDIAN.

Comment (rohitpaulk) [reply compiled_starters/java/src/main/java/Main.java:23]: Yep, I think the initial intention here was to be more explicit - but yeah safe to remove now that you mention it

**Comment by rohitpaulk** [comment]
@andy1li can you take this over please? Make the required changes to both Kotlin & Java

**Comment by rohitpaulk** [comment]
Looks good - feel free to merge @andy1li

**Pull request merged**